### PR TITLE
Chi 1209: Encrypt s3 buckets in account terraform config

### PR DESCRIPTION
**Pull request by stephenhand**
Primary reviewer: @nickhurlburt 

## Description

* S3 buckets created as part of a new twilio account are encrypted by default
* Some accounts have been updated to better align with

### Checklist
- [X] Corresponding issue has been opened
- N/A New tests added
- N/A Strings are localized
- N/A Tested for chat contacts
- N/A Tested for call contacts

### Related Issues
CHI-1209

### Verification steps

Review the affected buckets in the S3 console in AWS

### CODE REVIEW

File: twilio-iac/terraform-modules/aws/default/main.tf [hunk 1]
```diff
@@ -59,6 +67,25 @@ resource "aws_s3_bucket_public_access_block" "chat" {
...
+    allowed_origins = ["https://flex.twilio.com"]
+    expose_headers = []
+  }
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "chat" {
+  bucket = aws_s3_bucket.chat.bucket
+  rule {
+    apply_server_side_encryption_by_default {
+      sse_algorithm     = "AES256"
```
Comment (nickhurlburt) [inline twilio-iac/terraform-modules/aws/default/main.tf:84]: Do we know if this is with S3 keys or KMS?

**Review by nickhurlburt — APPROVED** [review]
There's one issue at Line 79 of twilio-iac/kidshelpphone-staging/main.tf that looks wrong, which does need addressing. All the other things are questions or nits.

File: twilio-iac/kidshelpphone-staging/main.tf [hunk 2]
```diff
@@ -73,6 +75,8 @@ module survey {
   source = "../terraform-modules/survey/default"
   helpline = var.helpline
   flex_task_assignment_workspace_sid = module.taskRouter.flex_task_assignment_workspace_sid
+  custom_task_routing_filter_expression = "isSurveyTask==true"
+  custom_task_routing_survey_queue_target_filter_expression =
```
[the remark this answers is not in the document]
Comment (stephenhand) [reply twilio-iac/kidshelpphone-staging/main.tf:79]: Thanks for catching that

File: twilio-iac/terraform-modules/aws/default/main.tf [hunk 3]
```diff
@@ -59,6 +67,25 @@ resource "aws_s3_bucket_public_access_block" "chat" {
...
+  bucket = aws_s3_bucket.chat.bucket
+  cors_rule {
+    allowed_headers = ["*"]
+    allowed_methods = ["GET", "POST", "PUT"]
+    allowed_origins = ["https://flex.twilio.com"]
+    expose_headers = []
+  }
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "chat" {
```
[the remark this answers is not in the document]
Comment (stephenhand) [reply twilio-iac/terraform-modules/aws/default/main.tf:80]: Unfortunately you specify a single bucket for the configuration for this resource, it doesn't seem to be intended for reuse, same with the other s3 config resources

[the remark this answers is not in the document]
Comment (GPaoloni) [reply twilio-iac/terraform-modules/aws/default/main.tf:80]: Maybe the shared parts of the configuration can be described in template files? Not sure if it's worth the effort tho, just throwing in an idea.

**Review by GPaoloni — APPROVED** [review]
Looks good!

**Pull request merged**